cookbook/grid_pad.py

```diff
@@ -0,0 +1,79 @@
+"""
+Gridding: Pad gridded data
+"""
+
+from fatiando import mesher,gridder
+from fatiando.gravmag import prism
+from fatiando.vis import mpl
+import numpy as np
+
+# Generate gridded data
+shape = (101,172)
+x,y,z = gridder.regular((-5000, 5000, -5000, 5000), shape, z=-150)
+model = [mesher.Prism(-4000, -3000, -4000, -3000, 0, 2000, {'density':1000}),
+        mesher.Prism(-1000, 1000, -1000, 1000, 0, 2000, {'density':-900}),
+        mesher.Prism(2000, 4000, 3000, 4000, 0, 2000, {'density':1300})]
+gz = prism.gz(x, y, z, model)
+gz = gz.reshape(shape)
+
+# Pad arrays with all the padding options
+pads = []
+xy = np.zeros((len(x),2))
+xy[:,0] = x[:]
+xy[:,1] = y[:]
```

Comment:
Sure, a list is fine.  There was a reason I did it this way--couldn't tell
you what that reason was now though...

It returns a list of coordinate arrays so no reason to not expect a list
coming in.  Wonder why I did it that way?

Anyway, I'll start a \todo list and add this to it.

On Fri, Dec 4, 2015 at 12:04 PM, Leonardo Uieda <notifications@github.com>
wrote:

> In cookbook/grid_pad.py
> <https://github.com/fatiando/fatiando/pull/239#discussion_r46718529>:
>
> > +import numpy as np
> > +
> > +# Generate gridded data
> > +shape = (101,172)
> > +x,y,z = gridder.regular((-5000, 5000, -5000, 5000), shape, z=-150)
> > +model = [mesher.Prism(-4000, -3000, -4000, -3000, 0, 2000, {'density':1000}),
> > +        mesher.Prism(-1000, 1000, -1000, 1000, 0, 2000, {'density':-900}),
> > +        mesher.Prism(2000, 4000, 3000, 4000, 0, 2000, {'density':1300})]
> > +gz = prism.gz(x, y, z, model)
> > +gz = gz.reshape(shape)
> > +
> > +# Pad arrays with all the padding options
> > +pads = []
> > +xy = np.zeros((len(x),2))
> > +xy[:,0] = x[:]
> > +xy[:,1] = y[:]
>
> @drandykass <https://github.com/drandykass> the handling of the
> coordinate arrays seems a bit too complicated. Wouldn't it be better to
> pass along (and return) a list of coordinate arrays? That would also avoid
> the memory copying involved in creating a new 2D array (the list holds only
> references to the arrays, not their values).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fatiando/fatiando/pull/239/files#r46718529>.
>



-- 
------------------------------------------------------------------------------------
Dum inter homines sumus, colamus humanitatem.
------------------------------------------------------------------------------------